python-package/pyproject.toml

```diff
@@ -57,26 +57,30 @@ changelog = "https://github.com/microsoft/LightGBM/releases"
 # start:build-system
 [build-system]
 
-requires = ["scikit-build-core>=0.4.4"]
+requires = ["scikit-build-core>=0.9.3"]
 build-backend = "scikit_build_core.build"
 
 # based on https://github.com/scikit-build/scikit-build-core#configuration
 [tool.scikit-build]
 
-cmake.minimum-version = "3.18"
-ninja.minimum-version = "1.11"
+cmake.version = ">=3.18"
+ninja.version = ">=1.11"
 ninja.make-fallback = true
 cmake.args = [
     "-D__BUILD_FOR_PYTHON:BOOL=ON"
 ]
 cmake.verbose = false
 cmake.build-type = "Release"
+cmake.targets = ["_lightgbm"]
```

Comment:
This ensures that only the shared library is built, and not other targets like the CLI or headers.

`scikit-build-core` didn't support that back when we did #5759, which is why this exists:

https://github.com/microsoft/LightGBM/blob/88cec4776e621ac93f9ba03aa0015035570545da/CMakeLists.txt#L54-L60

I'd like to leave that in `CMakeLists.txt` as extra protection for now. It's only a tiny bit of complexity in exchange for protection against some potentially-impactful side effects (like `pip install` of an sdist placing LightGBM's headers somewhere outside of where `pip` manages files).